docs/custom-domain/registry-setup.md

```diff
@@ -0,0 +1,94 @@
+  - custom data domain to access the registry content: `eastus-registry-data.contoso.com`
+    - Note that the custom data domain is region specific. For geo-replicated registries, each region should have its own custom data endpoint.
+
+  For each domain, you must prepare a single PEM formatted file containing the TLS private key and public certificate:
+  
+  ```
+  -----BEGIN PRIVATE KEY-----  
+  .....  
+  -----END PRIVATE KEY-----  
+  -----BEGIN CERTIFICATE-----  
+  .....  
+  -----END CERTIFICATE-----
+  ```
+  
+  For example, using [openssl](https://github.com/openssl/openssl):
+  - Create a self-signed public cert and private key
+     - `openssl req -nodes -x509 -newkey rsa:4096 -keyout container-registry.contoso.com.key.pem -out container-registry.contoso.com.cert.pem -days 365 -subj '/CN=container-registry.contoso.com/O=Contoso./C=US'`
+  - Create a single file containing both the public certificate and private key
+     - `cat container-registry.contoso.com.key.pem >> container-registry-contoso-com-pem`
+     - `cat container-registry.contoso.com.cert.pem >> container-registry-contoso-com-pem`
+  
+  Azure Key Vault allows you to [create certificates](https://docs.microsoft.com/azure/key-vault/certificate-scenarios) signed with a Certificate Authority.
+  
+## Prepare your existing registry
+We will enable two features on your registry that are currently in preview:
+- Data Endpoints:\
+  This feature provides a dedicated endpoint for downloading content from your registry. If you have a registry in East US, on enabling this feature, a data endpoint is automatically created for you: `eastus.data.myregistry.azurecr.io`
+  
+- ACR Managed Identities:\
+  Managed Identities (MI) provide a mechanism to associate an Azure Active Directory identity with your registry, while relieving you of the burden of managing credentials. To learn more, see the documentation [here](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).\
+ ACR supports both user assigned and system assigned MI. If you choose to enable user assigned MI for your registry, create one first before proceeding. Instructions [here](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal).
+
+### Enable preview features
+1. `az login`
+2. `az account set -s <subscription-id-or-name> `
+3. `apiVersion="api-version=2019-12-01-preview"`
+4. `mgmtEndpoint=$(az cloud show --query endpoints.resourceManager -o tsv)`
+5. `accessToken=$(az account get-access-token --query accessToken -o tsv)`
+6. `registryResourceId=$(az acr show -n myregistry --query id -o tsv)`
+7. You can either enable a system assigned MI, or a user assigned MI, or both for your registry. We recommend using a system assigned MI to enable advanced scenarios with virtual networks that are [called out below](#enhanced-security-with-virtual-networks). Do one of the following:
```

Comment:
Do you want to call out the advanced scenario since you state it's not supported at the moment?

Reply:
I've rephrased this a little. Does this make sense?